src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,76 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String DELIMITER = ",";
+    private static final String REGEX = "\\W+";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = readFile(fromFileName);
+        String report = generateReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    private int[] readFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            int supplyResult = 0;
+            int buyResult = 0;
+            while ((line = reader.readLine()) != null) {
+                String[] data = line.split(REGEX);
+                switch (data[OPERATION_TYPE_INDEX]) {
+                    case SUPPLY:
+                        supplyResult += Integer.parseInt(data[AMOUNT_INDEX]);
+                        break;
+                    case BUY:
+                        buyResult += Integer.parseInt(data[AMOUNT_INDEX]);
+                        break;
+                    default:
+                        break;
+                }
+            }
+            return new int[] {supplyResult, buyResult};
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read data from file", e);
```

Comment:
![image](https://github.com/mate-academy/jv-work-with-file/assets/49767049/931330ed-3006-43ff-9ef6-d204b6ba137a)
always include filename/file path to exceptions related to files